aws/resource_aws_guardduty_member.go

```diff
@@ -59,8 +84,63 @@ func resourceAwsGuardDutyMemberCreate(d *schema.ResourceData, meta interface{})
 	if err != nil {
 		return fmt.Errorf("Creating GuardDuty Member failed: %s", err.Error())
 	}
+
 	d.SetId(fmt.Sprintf("%s:%s", detectorID, accountID))
 
+	if !d.Get("invite").(bool) {
+		return resourceAwsGuardDutyMemberRead(d, meta)
+	}
+
+	imi := &guardduty.InviteMembersInput{
+		DetectorId:               aws.String(detectorID),
+		AccountIds:               []*string{aws.String(accountID)},
+		DisableEmailNotification: aws.Bool(d.Get("disable_email_notification").(bool)),
+		Message:                  aws.String(d.Get("invitation_message").(string)),
+	}
+
+	_, err = conn.InviteMembers(imi)
+	if err != nil {
+		return fmt.Errorf("Inviting GuardDuty Member failed: %s", err.Error())
+	}
+
+	// wait until e-mail verification finishes
+	if err = resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		input := guardduty.GetMembersInput{
+			DetectorId: imi.DetectorId,
+			AccountIds: imi.AccountIds,
+		}
+
+		log.Printf("[DEBUG] Reading GuardDuty Member: %s", input)
+		gmo, err := conn.GetMembers(&input)
+
+		if err != nil {
+			if isAWSErr(err, guardduty.ErrCodeBadRequestException, "The request is rejected because the input detectorId is not owned by the current account.") {
+				log.Printf("[WARN] GuardDuty detector %q not found, removing from state", d.Id())
+				d.SetId("")
+				return nil
+			}
+			return resource.RetryableError(fmt.Errorf("Reading GuardDuty Member '%s' failed: %s", d.Id(), err.Error()))
+		}
+
+		if gmo.Members == nil || (len(gmo.Members) < 1) {
+			log.Printf("[WARN] GuardDuty Member %q not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		member := gmo.Members[0]
+		d.Set("relationship_status", member.RelationshipStatus)
```

Comment:
One last extraneous `d.Set()` during create -- will remove and instead directly reference the output attribute from the API. I'll fix this on merge.